app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        print(f"Animal __init__ called with: {name}, health: {health}")
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+# Use
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
Code Efficiency: You can change the `bool` value in one line. Instead of using an if-else statement, you can simply use the 'not' keyword. This would make the code cleaner and more efficient.